libs/execution/src/lib/constraints/executors/allowlist-constraint-executor.ts

```diff
@@ -16,8 +20,12 @@ export class AllowlistConstraintExecutor implements ConstraintExecutor {
       return false;
     }
 
-    const allowlist = context.getTextCollectionPropertyValue('allowlist');
-    const allowlistValues = allowlist.map((e) => e.value);
+    const allowlist = context.getExpressionCollectionPropertyValue('allowlist');
+    const allowlistValues = allowlist.map((expression) => {
```

Comment:
Imho could be written nicer if you split up the individual steps more:
 
```
  allowlist
    .map(evaluateExpression)
    .filter(value => typeof value === 'string')
    .includes(value)
```

Not needed of course but just personal preference.

Reply:
I'd like to keep the `assert` runtime check rather than silently filtering non-string values (which should never happen at this point). So I'll keep the current implementation.